### DISCUSSION TITLE
Need to access parent data and create child data based on parent data.

### DISCUSSION THREAD

**Discussion by shashank-brightness** [alpinejs/alpine#3718] [1. Help]
I am trying to access parent data where there is an array of objects on parent and then trying to create new entry of parent array based on some condition. 

Here `shops` is the data which has been registered via Alpine.data method before alpine starts i.e. on `alpine:init` event via the cdn way. The problem is I can;t add another child data whoch should be based on parent but only the specific entry like we used to find the entry from array in plain js ES-6 way. 

I thought it would work but when i debug and see alpine js tab it shows shop as undefined.

```
<div class="d-flex flex-column flex-grow-1"  x-data="shops">
      <div class="container container-xxxl map-shops" x-data="{
            shop: $data.shops.find(shop => shop.hasSomeCondition)
        }">
        <h3 x-text="shop.title"></h3>
      </div>
  </div>
```

however if I add template loop two times thus, not creating any child data (the ugly way imo) then it works perfectly. Please note that shops data has even the same named property `shops` which is array populated via api.

```
<div class="d-flex flex-column flex-grow-1"  x-data="shops">
  <div class="container container-xxxl map-shops">
      <template x-for="(shop,index) in shops" :key="shop.id">
          <template x-if="shop.title.includes('bla bla bla')">
            <h3 x-text="shop.title"></h3>
          </template>
      </template>
    </div>
</div>
```
and title is printed when it has something like condition.

Can you please help figure out why array find method is not working the first way? Also, can you please suggest how to debug that line where I added `$data.shops.find(shop => shop.hasSomeCondition)` in chrome debugger as alpine tab only allows us to see the property and not debug?

**Comment by ekwoka** [comment]
first way just do

```html
<div x-data="shops">
  <div x-data="{ shop: shops.find(shop => shop.hasSomeCondition) }">
    <h3 x-text="shop.title"></h3>
  </div>
</div>
```

you just don't need $data there.

the second one needs an element between the templates, btw to actually work properly.

if you only need one, why not just have a getter on the `shops` to begin with?

**Comment by shashank-brightness** [comment]
@ekwoka 
Thanks for your reply but unfortunately it doesn't work that way. 

I tried the same way as your answer before I posted here but then I got error saying `can't use find method of undefined`. Do you have any idea why i can't use find?

 With $data, at least I am not getting that find method error but also at the same time not getting `shop` any time. Please note that shops data is initially empty array which is populated as the data is fetched via api.
 
Strangely, If i use `newShops: $data.shops` in child then I get the newShops same as parent shops but then why can't use `find` then?

**Reply by ekwoka** [reply]
oh, the issue is the same name...

That would do it.

It should really just be a getter.

Why does this need a new x-data?

**Reply by shashank-brightness** [reply]
@ekwoka What do you mean by same name? I am keeping different name here for new property. 
i.e.  `shop: shops.find(shop => shop.hasSomeCondition) }` so in that way, shop should be just a single shop from shops.

I am kind of feel like its too much to loop through entire array just to get simple one element from it and print it's data via one more inner template node (`<template x-if="shop.title.includes('bla bla bla')">` in second example) because when we inspect we found that there are many template nodes created just to print single entry there. 

That's why I need to extract that element somewhere first (in child here) and then print,  and that I don't want to do in data itself but in child only. In short I should be able to do anything with shops once i get them (sort, reverse, find and so on ...) which i don't right now.

**Reply by ekwoka** [reply]
I mean, the `x-data="shops"` is same name as `shops` in the data

so within an `x-data` the defined data method wins

But still, jsut use a getter on the `shops` context.
